### PR TITLE
Added an end Date to IrImb DiagnosisExtractor

### PR DESCRIPTION
**Pull request by strayMat**
IrImb diagnosis should be handled carefully in studies. 

When used to characterize a patient, the IrImb diagnosis (ALD = Affection Longue Duree) should be active during the period of study. Thus we extract the endDate of ALD for IrImbDiagnosis.

### CODE REVIEW

File: src/main/scala/fr/polytechnique/cmap/cnam/etl/extractors/diagnoses/ImbDiagnosisExtractor.scala [hunk 1]
```diff
@@ -3,11 +3,16 @@
...
+
 import org.apache.spark.sql.{DataFrame, Row}
 import fr.polytechnique.cmap.cnam.etl.events.{Diagnosis, Event, ImbDiagnosis}
 import fr.polytechnique.cmap.cnam.etl.extractors.Extractor
 import fr.polytechnique.cmap.cnam.etl.sources.Sources
 import fr.polytechnique.cmap.cnam.util.datetime
+import fr.polytechnique.cmap.cnam.util.datetime.implicits._
+import fr.polytechnique.cmap.cnam.util.functions.makeTS
+
+import scala.util.Try
```
Comment (ysebiat) [inline src/main/scala/fr/polytechnique/cmap/cnam/etl/extractors/diagnoses/ImbDiagnosisExtractor.scala]: In the order of the imports, this should be first.

File: src/main/scala/fr/polytechnique/cmap/cnam/etl/extractors/diagnoses/ImbDiagnosisExtractor.scala [hunk 2]
```diff
@@ -39,11 +44,18 @@ trait ImbSource extends Serializable {
     row.getAs[Date](ColNames.Date).toTimestamp
   }
 
+  def getEventEnd(r: Row): Timestamp = {
```
Comment (ysebiat) [inline src/main/scala/fr/polytechnique/cmap/cnam/etl/extractors/diagnoses/ImbDiagnosisExtractor.scala]: Can you document this method and why it defaults to 2100 when there is a missing value.

**Review by ysebiat — CHANGES_REQUESTED** [review]
Some suggested changes.

File: src/main/scala/fr/polytechnique/cmap/cnam/etl/extractors/diagnoses/ImbDiagnosisExtractor.scala (hunk 1 above)
Comment (strayMat) [reply src/main/scala/fr/polytechnique/cmap/cnam/etl/extractors/diagnoses/ImbDiagnosisExtractor.scala]: Changed.

File: src/main/scala/fr/polytechnique/cmap/cnam/etl/extractors/diagnoses/ImbDiagnosisExtractor.scala (hunk 2 above)
Comment (strayMat) [reply src/main/scala/fr/polytechnique/cmap/cnam/etl/extractors/diagnoses/ImbDiagnosisExtractor.scala]: Changed to  return None if absent and documented irimb

**Review by ysebiat — APPROVED** [review]
LGTM.

**Pull request merged**